safekeeper/src/safekeeper.rs

```diff
@@ -91,6 +91,59 @@ impl TermHistory {
         }
         TermHistory(res)
     }
+
+    /// Find point of divergence between leader (walproposer) term history and
+    /// safekeeper. Arguments are not symmetrics as proposer history ends at
+    /// +infinity while safekeeper at flush_lsn.
+    /// C version is at walproposer SendProposerElected.
+    pub fn find_highest_common_point(
+        prop_th: &TermHistory,
+        sk_th: &TermHistory,
+        sk_wal_end: Lsn,
```

Comment:
Is`sk_wal_end` always >= than `sk_th.last().lsn`?

Reply:
Yes, it's a reasonable assert to add.

Reply:
added assert